internal/provider/role_mapping_aws.go

```diff
@@ -0,0 +1,161 @@
+package provider
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/tabular-io/tabular-sdk-go/tabular"
+	"github.com/tabular-io/terraform-provider-tabular/internal/provider/util"
+)
+
+var (
+	_ resource.Resource                = &roleMappingAWSResource{}
+	_ resource.ResourceWithConfigure   = &roleMappingAWSResource{}
+	_ resource.ResourceWithImportState = &roleMappingAWSResource{}
+)
+
+type roleMappingAWSResource struct {
+	client *util.Client
+}
+
+func NewRoleMappingAWSResource() resource.Resource {
+	return &roleMappingAWSResource{}
+}
+
+type roleMappingAWSResourceModel struct {
+	Id         types.String `tfsdk:"id"`
+	Name       types.String `tfsdk:"name"`
+	RoleId     types.String `tfsdk:"role_id"`
+	AWSRoleArn types.String `tfsdk:"aws_role_arn"`
+}
+
+func (r *roleMappingAWSResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*util.Client)
+}
+
+func (r *roleMappingAWSResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_role_mapping_aws"
+}
+
+func (r *roleMappingAWSResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```

Comment:
very much a nit:
```suggestion
func (r *AWSRoleMappingResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```

To match the AWS naming of AWSIAMPolicyDataSource? or should that one be switched to `IAMPolicyAWSDataSource`. Neutral on which ever, but would be nice to consistent in our naming when specifying it is cloud specifc